### ISSUE TITLE
Date sorting not working for collections

### ISSUE THREAD

**Issue by lorentrogers** [jekyll/jekyll#4687]
jekyll 3.1.2
Debian 8

I have a collection called portfolio, and the sort function does not work:

    {% for post in site.portfolio | sort: 'date', 'last' %}
    {% for post in site.portfolio | sort: 'date', 'first' %}

The dates are different for each post, but the result is the same.

**Comment by lorentrogers** [comment]
This may be intentional -- perhaps the documentation needs to be updated to explain how this is supposed to work. http://stackoverflow.com/questions/30933741/jekyll-cant-sort-collection-by-date/30946672?noredirect=1#comment59895077_30946672

**Comment by parkr** [comment]
@lorentrogers As of Jekyll 3.0, `date` is a special property for collection documents. It will be parsed. We should update the documentation to reflect this. Would you please submit a pull request?

Are you seeing problems like `Cannot compare Date to Time`?

**Comment by lorentrogers** [comment]
Good to know - I figured it was something like that. 

I'd be happy to submit a pull request. I assume this should be to the [site pages](https://github.com/jekyll/jekyll/tree/master/site)?

**Comment by DirtyF** [comment]
@lorentrogers yes, alternately, you can browse the online documentation and click the improve button

**Comment by jajourda** [comment]
+1

**Comment by envygeeks** [comment]
Please do not "+1" on issues, please use the "vote buttons" on a specific post.
